SimulationAgent/SimulationRunner.cs

```diff
@@ -290,7 +290,9 @@ private DeviceModel GetDeviceModel(string id, Services.Models.Simulation.DeviceM
             var modelDef = new DeviceModel();
             if (id.ToLowerInvariant() != DeviceModels.CUSTOM_DEVICE_MODEL_ID.ToLowerInvariant())
             {
-                modelDef = this.deviceModels.Get(id);
+                var task = this.deviceModels.GetAsync(id);
+                task.Wait(TimeSpan.FromMinutes(1));
```

Comment:
It will change the type of task to a `boolean` type

Reply:
make this method async, and on line 191 use .Wait and give it 30 secs. 

On line 191 if the method throws an exception, you will have to open the AggregateException to know what happened, and log it properly.  Make sure the app recovers from the exception (i.e. retry) otherwise the service might got into a failed mode and never recover